### ISSUE TITLE
RavenDB.Client = Missing method on unit tests

### ISSUE THREAD

**Issue by andre-castro-garcia** [ravendb/ravendb#16538]
Hello Everyone 👋🏻 

I'm not 100% sure why it happens (and I didn't have much time digging into this problem yet, but maybe this could help someone else facing the same issue) but I'm creating some unit tests for my .Net Core 6.0 API and I was facing an issue `MissingMethod` when running the RavenDB unit tests, this is strange behavior because when running the sample provided here (https://ravendb.net/docs/article-page/5.3/csharp/start/test-driver#unittest) everything works well, but when running the unit tests with multiple projects I was getting the issue 👇🏻 :

```
System.MissingMethodException : Method not found: 'System.Threading.Tasks.Task`1<System.Collections.Generic.IAsyncEnumerator`1<Raven.Client.Documents.Commands.StreamResult`1<!!0>>> Raven.Client.Documents.Session.IAsyncAdvancedSessionOperations.StreamAsync(System.Linq.IQueryable`1<!!0>, System.Threading.CancellationToken)'.
```

![image](https://github.com/ravendb/ravendb/assets/35741698/1d54e10e-2690-418e-9d05-39dafdadd661)

Just to give you guys a little bit more context, I have two projects in my solution **APIProject** and **TestProject**, the **APIProject** handles a bunch of endpoints and handlers (I'm also using Mediatr), the **TestProject** contains all unit tests for my application.

In the **APIProject** I had only a reference for **RavenDB.DependencyInjection**, this is because I'm injecting the `IDocumentStore` in my controllers at runtime and everything is working as expected, in the **TestProject** I added the reference for **RavenDB.TestDriver** to be able to open an in-memory instance of RavenDB to test some queries, and then when testing the controllers I got the issue above ☝🏻.
I was expecting the reference for `RavenDB.Client` to be inferred and everything run well but seems it is not what is happening, the message I got complains about failure to find `StreamAsync`. method. 😕 

So, long story short, to fix this issue just added the reference to `RavenDB.Client` explicitly in the **APIProject** and everything ran well since then, but the question still remains, why my code was working as expected when deployed but failing just for running the unit tests?

Any updates regarding this issue will be posted here. 👍🏻

**Comment by ppekrol** [comment]
Hi @andre-castro-garcia 

TBH it is really hard to say what happened there, I suspect that maybe the dependency resolution didn't work as you would expect in the test project, for example I see that `RavenDB.DependencyInjection` depends on RavenDB.Client >= 5.0.0 and IIRC NuGet takes minimal version and in other projects you used probably latest 5.4 so the deployment of your project worked fine, but the test project resolved the library to the wrong version.
